arctic/store/_pickle_store.py

```diff
@@ -54,12 +56,16 @@ def write(self, arctic_lib, version, symbol, item, previous_version):
         collection = arctic_lib.get_top_level_collection()
         # Try to pickle it. This is best effort
         version['blob'] = _MAGIC_CHUNKED
-        pickled = lz4.compressHC(cPickle.dumps(item, protocol=cPickle.HIGHEST_PROTOCOL))
+        try:
+            pickled = lz4.compressHC(cPickle.dumps(item, protocol=cPickle.HIGHEST_PROTOCOL))
+        except OverflowError:
```

Comment:
@bmoscon has hit the nail on the head with why we did it this way. The original mmap storage engine fragmented badly, so identically sized chunks were an advantage for the pickle store.  WT doesn't do this, so this is less of an issue.

I think compression buys us a lot, so lets fix this by compressing the chunks as @richardbounds suggests.

Also agree that this verged on *too much* data in a pickled item.  Mongo have always been surprised by how much data we're pushing into the database...  That said I see the argument that there shouldn't be an artificial limit here.